### PR TITLE
Datadog Webhook

### PR DESCRIPTION
**Pull request by PeyGis**
# Description

Added integration docs for Datadog and Port using webhook

## Added docs pages

docs/build-your-software-catalog/sync-data-to-catalog/webhook/examples/datadog.md

### CODE REVIEW

File: docs/build-your-software-catalog/sync-data-to-catalog/webhook/examples/resources/datadog/_example_datadog_webhook_configuration.mdx [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  "mappings": [
+    {
+      "blueprint": "datadogAlert",
+      "entity": {
+        "identifier": ".body.alert_id | tostring",
+        "title": ".body.title",
+        "properties": {
+          "url": ".body.event_url",
+          "message": ".body.message",
+          "event_type": ".body.event_type",
```
Comment (MPTG94) [inline docs/build-your-software-catalog/sync-data-to-catalog/webhook/examples/resources/datadog/_example_datadog_webhook_configuration.mdx:16]:
```suggestion
          "eventType": ".body.event_type",
```

File: docs/build-your-software-catalog/sync-data-to-catalog/webhook/examples/resources/datadog/_example_datadog_webhook_configuration.mdx [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+          "priority": ".body.priority",
+          "creator": ".body.creator",
+          "alert_metric": ".body.alert_metric",
```
Comment (MPTG94) [inline docs/build-your-software-catalog/sync-data-to-catalog/webhook/examples/resources/datadog/_example_datadog_webhook_configuration.mdx:19]:
```suggestion
          "alertMetric": ".body.alert_metric",
```

File: docs/build-your-software-catalog/sync-data-to-catalog/webhook/examples/resources/datadog/_example_datadog_webhook_configuration.mdx [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+          "alert_type": ".body.alert_type",
+          "tags": ".body.tags | split(\", \")"
+        },
+        "relations": {
+          "microservice": ".body.service"
```
Comment (MPTG94) [inline docs/build-your-software-catalog/sync-data-to-catalog/webhook/examples/resources/datadog/_example_datadog_webhook_configuration.mdx:24]: is the `service` key in the datadog request an array? you marked the relation as `many: true`

Comment (MPTG94) [reply docs/build-your-software-catalog/sync-data-to-catalog/webhook/examples/resources/datadog/_example_datadog_webhook_configuration.mdx:24]: After reviewing the documentation of variables in datadog, it looks that a single value will be sent, and not an array. So I think the relation should be `many: false`. Can you please verify?

**Review by MPTG94 — CHANGES_REQUESTED** [review]
please make changes according to comments

File: docs/build-your-software-catalog/sync-data-to-catalog/webhook/examples/resources/datadog/_example_datadog_webhook_configuration.mdx (hunk 3 above)
Comment (PeyGis) [reply docs/build-your-software-catalog/sync-data-to-catalog/webhook/examples/resources/datadog/_example_datadog_webhook_configuration.mdx:24]: Yes that's correct

File: docs/build-your-software-catalog/sync-data-to-catalog/webhook/examples/resources/datadog/_example_datadog_webhook_configuration.mdx [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+        }
+      }
+    }
+  ],
+  "enabled": true,
+  "security": {}
```
[the remark this answers is not in the document]
Comment (PeyGis) [reply docs/build-your-software-catalog/sync-data-to-catalog/webhook/examples/resources/datadog/_example_datadog_webhook_configuration.mdx:30]: Unlike the other platforms that provide a mechanism to verify the webhook payload, there isn't much information on how to achieve the same on Datadog's documentation. A substitute will be to leverage their **Custom Header** section when creating the webhook.
image

From the image above, the user can hash a webhook secret using a hashing algorithm such as the sha256. This value is then passed to the custom header as a property in Datadog. The user can then provide the security configuration in Port as follows:

  ```
  "security": {
    "secret": "WEBHOOK_SECRET_IN_PLAIN_TEXT",
    "signatureHeaderName": "X-Port-Signature",
    "signatureAlgorithm": "sha256"
  }
```

[the remark this answers is not in the document]
Comment (MPTG94) [reply docs/build-your-software-catalog/sync-data-to-catalog/webhook/examples/resources/datadog/_example_datadog_webhook_configuration.mdx]: Then let's just skip the security section for this doc

[the remark this answers is not in the document]
Comment (PeyGis) [reply docs/build-your-software-catalog/sync-data-to-catalog/webhook/examples/resources/datadog/_example_datadog_webhook_configuration.mdx:29]: sure, I've removed it then

**Review by MPTG94 — APPROVED** [review]
LGTM

**Pull request merged**